src/pages/home/ReportScreen.js

```diff
@@ -113,7 +113,7 @@ const defaultProps = {
  * @returns {String}
  */
 function getReportID(route) {
-    return route.params.reportID.toString();
+    return lodashGet(route, 'params.reportID', null);
```

Comment:
The main reason is to show the skeleton while the `reportID` is not loaded yet.
I made this change to allow `ReportScreen` to use `null` as `reportID`, which then is used to display the skeleton.
This approach was very convenient, because `ReportScreen` already implements skeleton. But, as you say, it may lead to some bugs with Onyx. I couldn't find in the docs option, to 'delay' the Onyx, until we have all required props.

Another solution might be, to add skeletons (when `reportID` is still loading) to the `ReportScreenWrapper`. If we want it to look exactly the same as it was before the navigation refactors, we would have to reimplement skeletons and add a header and footer there as well. It would be copying the same work as in `ReportScreen`, and would be poorly maintainable.

To sum up, I can:
- somehow use `ReportScreen` to display skeletons without the `reportID`,
- display the skeletons in the `ReportScreenWrapper`.

In my opinion, the first option would be cleaner and more maintainable. But I'm not sure if this is possible to work with `withOnyx`.

Reply:
The main reason is to show the skeleton while the `reportID` is not loaded yet (in the wrapper).
I made this change to allow the  `ReportScreen` to use `null` as `reportID`, which then is used to display the skeleton.

This approach is very convenient because `ReportScreen` already implements a skeleton. But, as you say, it may lead to some bugs with Onyx. 

I created a stripped-down version of `ReportScreen` when we don't know the `reportID` yet. It is used in `ReportScreenWrapper`.

---

Another solution might be, to add skeletons (when reportID is still loading) to the ReportScreenWrapper. But I don't think the first approach is better.